test/build/build.test.js

```diff
@@ -1,113 +0,0 @@
-const STRIP_CSS_MODULES = /_[\d\w]{5}_[\d]{1,3}/g;
-
-const ALREADY_TESTED = [
-  'base-url',
-  'base-url-homepage',
-  'base-url-remote',
-  'bugfix-named-import',
-  'cdn',
-  'config-extends-plugins',
-  'config-external-package',
-  'config-instantiated-object',
-  'config-js-file',
-  'config-mount',
-  'config-out',
-  'config-out-flag',
-  'config-treeshake',
-  'custom-modules-dir',
-  'entrypoint-ids',
-  'html-environment-variables',
-  'package-bootstrap',
-  'package-tippy-js',
-  'preload-css',
-];
-
-/** format diffs to be meaningful */
-function format(stdout) {
-  return stdout
-    .replace(STRIP_REV, '?rev=XXXXXXXXXX')
-    .replace(STRIP_CHUNKHASH, '$1XXXXXXXX$2')
-    .replace(STRIP_WHITESPACE, '')
-    .replace(STRIP_ROOTDIR, (_, p1, p2) => {
-      return `/HOME${(p1 + p2).replace(/\\{1,2}/g, '/')}`;
-    });
-}
-
-describe('snowpack build', () => {
-  for (const testName of readdirSync(__dirname)) {
-    if (testName === 'node_modules' || testName === '__snapshots__' || testName.includes('.')) {
-      continue;
-    }
-
-    // skip folders that have their own tests
-    if (ALREADY_TESTED.includes(testName)) continue;
-
-    it(testName, () => {
-      const cwd = path.join(__dirname, testName);
-      const relativePath = cwd.replace(process.cwd() + '/', '');
-
-      if (!existsSync(path.join(cwd, 'package.json'))) {
-        console.warn(
-          '%s folder has no package.json file, it is likely a leftover folder from a deleted test. You can remove the folder with `git clean -xdf %s`',
-          relativePath,
-          relativePath,
-        );
-
-        return;
-      }
-
-      // build test
-      const actual = path.join(cwd, 'build');
-
-      // Test That all files match
-      const allFiles = glob.sync(`**/*`, {
-        ignore: ['**/common/**/*'],
-        cwd: actual,
-        nodir: true,
-      });
-
-      if (allFiles.length === 0) {
-        throw new Error('Empty build directory!');
-      }
-
-      const snapshotFile = path.join(cwd, '__snapshots__');
-      expect(allFiles.map((f) => f.replace(/\\/g, '/'))).toMatchSpecificSnapshot(
-        snapshotFile,
-        'allFiles',
-      );
-
-      // If any diffs are detected, we'll assert the difference so that we get nice output.
-      for (const entry of allFiles) {
-        if (
-          entry.endsWith('.css') ||
-          entry.endsWith('.html') ||
-          entry.endsWith('.js') ||
-          entry.endsWith('.json') ||
-          entry.endsWith('.map')
-        ) {
-          let f1 = readFileSync(path.resolve(actual, entry), {encoding: 'utf8'});
-          // Add special handling for CSS module hashes
-          if (entry.includes('.module.css')) {
-            f1 = f1.replace(STRIP_CSS_MODULES, '_XXXXX_XX');
-          }
-          expect(format(f1)).toMatchSpecificSnapshot(
-            snapshotFile,
-            `build/${entry.replace(/\\/g, '/')}`,
-          );
-        }
-      }
-    });
-  }
-});
```

Comment:
Bye bye, `build.test.js`. We don‘t need this anymore, now that the individual tests have been broken out.

It also means we don‘t need any global RegEx to strip out all the different line endings for Unix vs Windows, ANSI colors, backslashes, and other weirdness. Or, if that is needed in a test, it can be handled case-by-case rather than trying to manage The One RegEx To Rule Them All™

Reply:
I'd hate to lose all of these useful helpers to the sands of time... any way that we could add a link to this commit somewhere in case we need to come back and re-add some of these? (maybe in test-utils?)

Reply:
yup, just needed to grab the "strip whitespace" regex to handle some mac vs. windows whitespace issues.

Reply:
Adding to `test-utils` (and updating the place where you included it)